esphome/components/captive_portal/__init__.py

```diff
@@ -38,3 +57,6 @@ async def to_code(config):
         cg.add_library("WiFi", None)
     if CORE.is_esp8266:
         cg.add_library("DNSServer", None)
+
+    if config.get(CONF_KEEP_USER_CREDENTIALS, False):
+        cg.add_define("USE_WIFI_USER_CREDENTIALS")
```

Comment:
Would just like the define to be named from the component that actually creates it.
```suggestion
        cg.add_define("USE_CAPTIVE_PORTAL_KEEP_USER_CREDENTIALS")
```

Reply:
Okay, done.